lib/container_orchestrator.rb

```diff
@@ -8,22 +8,41 @@ class ContainerOrchestrator
   TOKEN_FILE   = "/run/secrets/kubernetes.io/serviceaccount/token".freeze
   CA_CERT_FILE = "/run/secrets/kubernetes.io/serviceaccount/ca.crt".freeze
 
+  class_attribute :cached_deployment, :default => Hash.new({})
+
   def self.available?
     File.exist?(TOKEN_FILE) && File.exist?(CA_CERT_FILE)
   end
 
   def scale(deployment_name, replicas)
-    kube_apps_connection.patch_deployment(deployment_name, { :spec => { :replicas => replicas } }, my_namespace)
+    patch_deployment(deployment_name, { :spec => { :replicas => replicas } })
+  end
+
+  def patch_deployment(deployment_name, data)
+    _log.info("deployment_name: #{deployment_name}, data: #{data.inspect}")
+    kube_apps_connection.patch_deployment(deployment_name, data, my_namespace).tap do
+      cache_deployment(deployment_name, data, :merge)
+    end
   end
 
   def create_deployment(name)
     definition = deployment_definition(name)
     yield(definition) if block_given?
-    kube_apps_connection.create_deployment(definition)
+    kube_apps_connection.create_deployment(definition).tap { cache_deployment(name, definition) }
```

Comment:
With both `create_deployment` and `patch_deployment`, we only `cache_deployment` if it doesn't raise an exception.  This means that if you have an invalid value, such as a `limit` lower than the `request`, it could get rejected by the API.  I'll need to add some exception handling for this situation.

Reply:
This is a comment on the caching on the patching deployment side.  It's been moved to the server's monitoring side so this comment isn't relevant anymore.